<launch>
  <arg name="launch_driver" default="true" />

  <arg name="pcap"  default=""/>
  <arg name="device_ip" default="192.168.1.201"/>
  <arg name="lidar_port"  default="2368"/>
  <arg name="gps_port"  default="10111"/>
  <arg name="scan_phase"  default="0.0"/>
  <arg name="return_mode" default="Dual"/>
  <arg name="dual_return_distance_threshold" default="0.1"/>
  <arg name="model" default="PandarQT"/>
  <arg name="frame_id" default="pandar"/>
  <arg name="topic" default="pointcloud_raw"/>
  <arg name="topic_ex" default="pointcloud_raw_ex"/>
  <arg name="driver_node" default="pandar_driver"/>
  <arg name="cloud_node" default="pandar_cloud"/>
  <arg name="calibration" default="$(find-pkg-share pandar_pointcloud)/config/qt_awsim.csv"/>

  <!-- pandar driver -->
  <node pkg="pandar_driver" exec="pandar_driver_node" name="$(var driver_node)" output="screen" >
    <param name="pcap" value="$(var pcap)"/>
    <param name="device_ip"  value="$(var device_ip)"/>
    <param name="lidar_port" value="$(var lidar_port)"/>
    <param name="gps_port"   value="$(var gps_port)"/>
    <param name="scan_phase" value="$(var scan_phase)"/>
    <param name="model"      value="$(var model)"/>
    <param name="frame_id"   value="$(var frame_id)"/>
  </node>

  <!-- pandar cloud -->
  <node pkg="pandar_pointcloud" exec="pandar_cloud_node" name="$(var cloud_node)" output="screen" >
    <remap from="pandar_points" to="$(var topic)" />
    <remap from="pandar_points_ex" to="$(var topic_ex)" />
    <param name="scan_phase"  value="$(var scan_phase)"/>
    <param name="return_mode" value="$(var return_mode)"/>
    <param name="dual_return_distance_threshold" value="$(var dual_return_distance_threshold)"/>
    <param name="model"       value="$(var model)"/>
    <param name="device_ip"   value="$(var device_ip)"/>
    <param name="calibration" value="$(var calibration)"/>
  </node>
</launch>
